board/common/rootfs/usr/libexec/infix/hw-wait

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+# (Ab)use the kernel's device link subsystem to detect consumer side
+# devices that may be very slow to probe (looking at you, mv88e6xxx!).
+
+ident=$(basename "$0")
+
+report()
+{
+    if [ -r /tmp/$ident ]; then
+	logger -k -p user.$1 -t "$ident" "Waited for slow devices:"
+	sort /tmp/$ident | uniq -c | logger -k -p user.$1 -t "$ident"
+    fi
+
+    rm -f /tmp/$ident
+}
+
+for i in $(seq 50); do
+    again=
+
+    for dl in /sys/class/devlink/*; do
+    	[ -r "$dl/status" ] || continue
+
+    	status=$(cat "$dl/status")
+	if [ "$status" = "consumer probing" ]; then
+	    basename $(readlink "$dl/consumer") >>/tmp/$ident
```

Comment:
Also here, around the `basename` argument, which might be slightly more relevant since `$dl` *may* contain spaces.  Again, still a minor comment.

Reply:
Thanks! I finally took your advise and set up flymake :sweat_smile: